app/src/main/java/com/example/recipeapp/EditPreferencesActivity.java

```diff
@@ -0,0 +1,129 @@
+package com.example.recipeapp;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
+
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.widget.ArrayAdapter;
+
+import com.example.recipeapp.Adapters.AutoCompleteAdapter;
+import com.example.recipeapp.Adapters.CheckboxAdapter;
+import com.example.recipeapp.Adapters.StepsAdapter;
+import com.example.recipeapp.CustomViews.MultipleSpinnerItem;
+import com.example.recipeapp.Models.Ingredient;
+import com.example.recipeapp.Models.Settings;
+import com.example.recipeapp.Retrofit.RecipeApi;
+import com.example.recipeapp.Retrofit.RetrofitClientInstance;
+import com.example.recipeapp.databinding.ActivityEditPreferencesBinding;
+import com.google.android.flexbox.FlexDirection;
+import com.google.android.flexbox.FlexWrap;
+import com.google.android.flexbox.FlexboxLayoutManager;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import retrofit2.Call;
+
+public class EditPreferencesActivity extends AppCompatActivity {
+
+    public static final String DIET = "diet";
+    public static final String CUISINES = "cuisines";
+    public static final String INTOLERANCES = "intolerances";
+    public static final String BANNED = "banned";
+    ActivityEditPreferencesBinding binding;
+    StepsAdapter banAdapter;
+    List<String> banned = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityEditPreferencesBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        binding.spinnerDiet.setAdapter(
+                new ArrayAdapter<String>(
+                        this,
+                        android.R.layout.simple_list_item_1,
+                        getResources().getStringArray(R.array.diets)
+                )
+        );
+
+        binding.spinnerDiet.setSelection(getDietIndex());
+
+        ArrayList<MultipleSpinnerItem> cuisinesVO = new ArrayList<>();
```

Comment:
qq, what `VO` means?

Reply:
View Object, but I renamed it for clarity